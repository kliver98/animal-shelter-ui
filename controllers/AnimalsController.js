import axios from 'axios';

export const AnimalController = {
    register(animal) {
        return axios({
            method: 'POST',
            baseURL: process.env.API,
            url: `animals`,
            data: animal,
        })
    },
    list() {
        return axios({
            method: 'GET',
            baseURL: process.env.API,
            url: 'animals'
        });
    },
    delete(name) {
        return axios({
            method: 'DELETE',
            baseURL: process.env.API,
            url: `animals/${name}`,
        });
    },
    getAnimal(name) {
        return axios({
            method: 'GET',
            baseURL: process.env.API,
            url: `animals/${name}`,
        });
    },
    updateAnimal(name) {
        return axios({
            method: 'PUT',
            baseURL: process.env.API,
            url: `animals/${name}`,
        });
    }
}

