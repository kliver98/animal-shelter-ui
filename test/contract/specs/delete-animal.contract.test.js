import { provider } from '../config/init-pact';
import { AnimalController } from '../../../controllers';
import { Matchers } from '@pact-foundation/pact';

const animal_name = 'prueba';

describe('Given an animal service', () => {
    beforeAll(async() => {
        await provider.setup();
    });

    describe('When a request to delete an animal is made', () => {
        beforeAll(async() => {
            await provider.addInteraction({
                state: 'backend service is up',
                uponReceiving: 'a request to delete an animal',
                withRequest: {
                    method: 'DELETE',
                    path: '/animals/'+animal_name
                },
                willRespondWith: {
                    status: 200,
                    body: Matchers.eachLike({
                        message: Matchers.string('Animal deleted successfully'),
                    }, {min: 1})
                }
            });
        });

        test('Then it should return message confirming the animal deleted', async() => {
            const response = await AnimalController.delete(animal_name);
            expect(response.data).toMatchSnapshot();

            await provider.verify();
        })

    });

    afterAll(async () => {
        await provider.finalize();
    });
});
