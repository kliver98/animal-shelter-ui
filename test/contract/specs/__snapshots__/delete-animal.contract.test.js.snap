// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an animal service When a request to delete an animal is made Then it should return message confirming the animal deleted 1`] = `
Array [
  Object {
    "message": "Animal deleted successfully",
  },
]
`;
